### PR TITLE
Refactor codes

### PR DESCRIPTION
**Pull request by vtdat**
* Fix typos
* Reformat logs
* Refactor scalers to use registry in utilities
* Release scalers mutex before jumping to next loop

### CODE REVIEW

File: pkg/autoscaler/manager.go [hunk 1]
```diff
@@ -182,22 +182,25 @@ func (m *Manager) getBackend(key string) (metrics.Backend, error) {
...
+						"name", i.Name, "err", err)
+					return
+				}
+				_, err = m.etcdcli.Put(context.Background(), i.Name, string(raw))
+				if err != nil {
+					level.Error(m.logger).Log("msg", "Error putting scaler object",
+						"name", i.Name, "err", err)
+					return
+				}
+				m.stopScaler(i.Name)
```
Comment (ntk148v) [inline pkg/autoscaler/manager.go]: Should return error or a log line if not Scalers.

File: pkg/autoscaler/manager.go [hunk 2]
```diff
@@ -110,33 +110,33 @@ func (m *Manager) Run(ctx context.Context) {
...
 		level.Debug(m.logger).Log("msg", "Ignoring scaler, another worker node takes it",
-			"scaler", id, "local", local, "worker", worker)
+			"scaler", name, "local", local, "worker", worker)
 		return
 	}
-	level.Info(m.logger).Log("msg", "Removing scaler", "id", id)
-	s.stop()
-	m.rgt.Delete(id)
+	level.Info(m.logger).Log("msg", "Removing scaler", "name", name)
+	s.Stop()
```
[the remark this answers is not in the document]
Comment (vtdat) [reply pkg/autoscaler/manager.go]: Yes, but the problem is that the function called is actually in the `Worker` interface which locates in `utils` package.

File: pkg/autohealer/healer.go [hunk 3]
```diff
@@ -302,10 +301,10 @@ func (h *Healer) Stop() {
...
 		return
 	}
-	level.Debug(h.logger).Log("msg", "Healer is stopping", "id", h.ID)
+	level.Debug(h.logger).Log("msg", "Healer is stopping", "id")
 	h.state = stateStopping
 	close(h.done)
 	<-h.terminated
 	h.state = stateStopped
-	level.Debug(h.logger).Log("msg", "Healer is stopped", "id", h.ID)
+	level.Debug(h.logger).Log("msg", "Healer is stopped", "id")
```
[the remark this answers is not in the document]
Comment (vtdat) [reply pkg/autohealer/healer.go]: sure, thanks.

File: pkg/autoscaler/manager.go (hunk 1 above)
Comment (vtdat) [reply pkg/autoscaler/manager.go]: i don't think it's needed but to be strict, sure.

File: pkg/autoscaler/manager.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ntk148v) [reply pkg/autoscaler/manager.go]: :man_facepalming: How about move Worker to module named `common`?

[the remark this answers is not in the document]
Comment (vtdat) [reply pkg/autoscaler/manager.go]: still outside of package and `stop` needs to be `Stop`

**Pull request merged**